src/services/pcn-iptables/src/Utils.cpp

```diff
@@ -479,6 +479,61 @@ bool Chain::portFromRulesToMap(
   return brk;
 }
 
+bool Chain::interfaceFromRulesToMap(
+        const uint8_t &type, std::map<uint16_t, std::vector<uint64_t>> &interfaces,
+        const std::vector<std::shared_ptr<ChainRule>> &rules,
+        Iptables& iptables) {
+  std::vector<uint16_t> dont_care_rules;
+
+  uint32_t rule_id;
+  uint16_t interface;
+
+  bool brk = true;
+
+  for (auto const &rule : rules) {
+    try {
+      rule_id = rule->getId();
+      interface = 0;
+      if (type == IN_TYPE) {
+        std::string interface_string = rule->getInIface();
+        interface = iptables.interfaceNameToIndex(interface_string);
+      } else {
+        std::string interface_string = rule->getOutIface();
+        interface = iptables.interfaceNameToIndex(interface_string);
+      }
+    } catch (std::runtime_error re) {
```

Comment:
Why just catching `std::runtime_error`, I think in this case using `...` could be better.

Reply:
In both `IN_TYPE` and `OUT_TYPE`, exception is raised by `getIn/OutInterface()` if this field is not set, and is raised by `interfaceNameToIndex()` if interface name is not valid.
If exception is raised in one of the previous case it means I have to consider current rule, as a don't care rule `*`.

This block can be replaced by a conditional, but this requires to change the pattern we use to handle this cases.
Do you think it is worth? What benefits can you see?

Reply:
After an offline talk with Mauricio, we agreed using exceptions for control flow handling is not the best practice.
Furthermore, this improvement will require many changes to the code, so it will be part of a  separate PR.